### PR TITLE
ESDT local burn enabled for cross chain operations

### PR DESCRIPTION
**Pull request by mariusmihaic**
Enabled ESDT local burn enabled for cross chain operations.
Added `CrossChainTokenChecker` interface to be used for local mint as well.

### CODE REVIEW

File: builtInFunctions/creator.go [hunk 1]
```diff
@@ -250,7 +253,21 @@ func (b *builtInFuncCreator) CreateBuiltInFunctionContainer() error {
...
 	}
 
-	newFunc, err = NewESDTLocalBurnFunc(b.gasConfig.BuiltInCost.ESDTLocalBurn, b.marshaller, globalSettingsFunc, setRoleFunc, b.enableEpochsHandler)
+	crossChainTokenCheckerHandler, err := NewCrossChainTokenChecker(b.selfESDTPrefix)
+	if err != nil {
+		return err
+	}
+
+	newFunc, err = NewESDTLocalBurnFunc(
+		ESDTLocalBurnFuncArgs{
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply builtInFunctions/creator.go]: Fixed it

**Review by sasurobert — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

File: builtInFunctions/esdtLocalBurn_test.go [hunk 2]
```diff
@@ -130,13 +168,15 @@ func TestEsdtLocalBurn_ProcessBuiltinFunction_ValueTooLong(t *testing.T) {
 	t.Parallel()
 
 	marshaller := &mock.MarshalizerMock{}
-	esdtRoleHandler := &mock.ESDTRoleHandlerStub{
+	args := createESDTLocalBurnArgs()
```
Comment (axenteoctavian) [inline builtInFunctions/esdtLocalBurn_test.go:171]: missed `args.Marshaller = marshaller`

File: builtInFunctions/esdtLocalBurn_test.go [hunk 3]
```diff
@@ -183,13 +223,15 @@ func TestEsdtLocalBurn_ProcessBuiltinFunction_ShouldWork(t *testing.T) {
...
```
Comment (axenteoctavian) [inline builtInFunctions/esdtLocalBurn_test.go:226]: missed `args.Marshaller = marshaller`

File: builtInFunctions/esdtLocalBurn_test.go [hunk 4]
```diff
@@ -236,15 +278,19 @@ func TestEsdtLocalBurn_ProcessBuiltinFunction_WithGlobalBurn(t *testing.T) {
 	t.Parallel()
 
 	marshaller := &mock.MarshalizerMock{}
-	esdtLocalBurnF, _ := NewESDTLocalBurnFunc(50, marshaller, &mock.GlobalSettingsHandlerStub{
+	args := createESDTLocalBurnArgs()
```
Comment (axenteoctavian) [inline builtInFunctions/esdtLocalBurn_test.go:281]: missed `args.Marshaller = marshaller`

**Review by axenteoctavian — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Pull request merged**